tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -171,7 +171,7 @@ function test_sync_always_sync_changes_to_modules_right_away() {
 		$this->sender->do_sync();
 
 		$synced_value = $this->server_replica_storage->get_callable( 'active_modules' );
-		$this->assertEquals( array( 'json-api' ), $synced_value );
+		$this->assertEquals( array( 1 => 'json-api' ), $synced_value );
```

Comment:
The fact that we have to change this test seems to indicate a regression. I might be misunderstanding, but I wouldn't expect for keys to start from `1` in a basic array with numeric keys.

Reply:
This was actually not meant as a fix to the failing test, just a quick & dirty change I made to see if other tests pass. You're right that array key starting with `1` indicates a regression, I will look into a proper fix.

Reply:
How did this end up landing?

Reply:
Yeah, would be nice to see what's going on here before 🚢

Reply:
> but I wouldn't expect for keys to start from 1 in a basic array with numeric keys.

This was happening since I introduced `array_diff` in `update_active_modules()` and array_diff preserves keys. I've fixed it in [72a8b5c](https://github.com/Automattic/jetpack/pull/12270/commits/72a8b5ca1323b2621dd3bc0496bcbf2eefed14a1) and [c3826bf](https://github.com/Automattic/jetpack/pull/12270/commits/c3826bf88dd8dbac8aab1f33205d8caf829670ac).

> Also, it seems that some unit test cases for the problems you described would be very useful - I'd really appreciate adding some.

I've added a new unit test that covers the case fixed by this PR.

cc @kraftbj @tyxla .